crates/client/src/client/schema/chain.rs

```diff
@@ -4,28 +4,127 @@ use crate::client::schema::{
+    pub gas_costs: GasCosts,
+}
+
+#[derive(cynic::QueryFragment, Debug)]
+#[cynic(schema_path = "./assets/schema.sdl")]
+pub struct TxParameters {
     pub max_inputs: U64,
     pub max_outputs: U64,
     pub max_witnesses: U64,
     pub max_gas_per_tx: U64,
-    pub max_script_length: U64,
-    pub max_script_data_length: U64,
-    pub max_storage_slots: U64,
+}
+
+impl From<TxParameters> for fuel_core_types::fuel_tx::TxParameters {
+    fn from(params: TxParameters) -> Self {
+        Self {
+            max_inputs: params.max_inputs.0,
+            max_outputs: params.max_outputs.0,
+            max_witnesses: params.max_witnesses.0,
+            max_gas_per_tx: params.max_gas_per_tx.0,
+        }
+    }
+}
+
+#[derive(cynic::QueryFragment, Debug)]
+#[cynic(schema_path = "./assets/schema.sdl")]
+pub struct PredicateParameters {
     pub max_predicate_length: U64,
     pub max_predicate_data_length: U64,
+    pub max_message_data_length: U64,
     pub max_gas_per_predicate: U64,
+}
+
+impl From<PredicateParameters> for fuel_core_types::fuel_tx::PredicateParameters {
+    fn from(params: PredicateParameters) -> Self {
+        Self {
+            max_predicate_length: params.max_predicate_length.0,
+            max_predicate_data_length: params.max_predicate_data_length.0,
+            max_message_data_length: params.max_message_data_length.0,
+            max_gas_per_predicate: params.max_gas_per_predicate.0,
+        }
+    }
+}
+
+#[derive(cynic::QueryFragment, Debug)]
+#[cynic(schema_path = "./assets/schema.sdl")]
+pub struct ScriptParameters {
+    pub max_script_length: U64,
+    pub max_script_data_length: U64,
+}
+
+impl From<ScriptParameters> for fuel_core_types::fuel_tx::ScriptParameters {
+    fn from(params: ScriptParameters) -> Self {
+        Self {
+            max_script_length: params.max_script_length.0,
+            max_script_data_length: params.max_script_data_length.0,
+        }
+    }
+}
+
+#[derive(cynic::QueryFragment, Debug)]
+#[cynic(schema_path = "./assets/schema.sdl")]
+pub struct ContractParameters {
+    pub contract_max_size: U64,
+    pub max_storage_slots: U64,
+}
+
+impl From<ContractParameters> for fuel_core_types::fuel_tx::ContractParameters {
+    fn from(params: ContractParameters) -> Self {
+        Self {
+            contract_max_size: params.contract_max_size.0,
+            max_storage_slots: params.max_storage_slots.0,
+        }
+    }
+}
+
+#[derive(cynic::QueryFragment, Debug)]
+#[cynic(schema_path = "./assets/schema.sdl")]
+pub struct FeeParameters {
     pub gas_price_factor: U64,
     pub gas_per_byte: U64,
-    pub max_message_data_length: U64,
-    pub chain_id: U64,
 }
 
+impl From<FeeParameters> for fuel_core_types::fuel_tx::FeeParameters {
+    fn from(params: FeeParameters) -> Self {
+        Self {
+            gas_price_factor: params.gas_price_factor.0,
+            gas_per_byte: params.gas_per_byte.0,
+        }
+    }
+}
+
+// macro_rules! include_from_impls {
```

Comment:
It should be simpler for you to use macro than copy paste it=)

Reply:
```
error[E0425]: cannot find value `builder` in this scope
   --> crates/client/src/client/schema/chain.rs:114:39
    |
114 |               $($field_vis $field_name: $field_type,)*
    |                                         ^^^^^^^^^^^ not found in this scope
...
128 | / include_from_impls! {
129 | | #[derive(cynic::QueryFragment, Debug)]
130 | | #[cynic(schema_path = "./assets/schema.sdl")]
131 | | pub struct GasCosts {
```

I'm getting this error when I try to use the macro. My best guess is that it doesn't like the attributes:
```
#[derive(cynic::QueryFragment, Debug)]
#[cynic(schema_path = "./assets/schema.sdl")]
```

Reply:
I modified the macro to include the cynic stuff so this isn't an issue.

Not the cleanest IMO, but it was the way to get the macro to work.